src/main/kotlin/com/openlattice/transporter/services/TransporterService.kt

```diff
@@ -0,0 +1,215 @@
+    }
+
+    /**
+     * Regular poll executed by [TransporterRunSyncTask]
+     */
+    fun pollOnce() {
+        val timer = pollTimer.startTimer()
+        val futures = this.transporterState.keys.map { entityTypeId ->
+            val relevantEntitySets = validEntitySets(entityTypeId)
+            val entitySetIds = relevantEntitySets.map { it.id }.toSet()
+            val partitions = partitions(entitySetIds)
+            val ft = transporterState.submitToKey(
+                    entityTypeId,
+                    TransporterPropagateDataEntryProcessor(relevantEntitySets, partitions).init(data)
+            ).toCompletableFuture()
+
+            entitySetIds.count() to ft
+        }
+        val setsPolled = futures.map { it.first }.sum()
+        // wait for all futures to complete.
+        val exception = MultiException()
+        futures.forEach { (_, f) ->
+            try {
+                f.get()
+            } catch(e: Exception) {
+                errorCount.inc()
+                exception.add(e)
+            }
+        }
+        val duration = timer.observeDuration()
+        logger.debug("Total poll duration time for {} entity sets in {} entity types: {} sec", setsPolled, futures.size, duration)
+        exception.ifExceptionThrow()
+    }
+
+    private fun refreshDataForEntitySet(entitySetId: UUID) {
+        val esCompletion = entitySets.getAsync( entitySetId )
+        val partitions = partitionManager.getEntitySetPartitions( entitySetId )
+        val refreshTimer = refreshTimer.startTimer()
+        esCompletion.thenCompose{ es ->
+            transporterState.submitToKey(es.entityTypeId,
+                    TransporterPropagateDataEntryProcessor(setOf(es), partitions).init(data)
+            )
+        }.whenCompleteAsync { _, throwable ->
+            refreshTimer.observeDuration()
+            if ( throwable != null ){
+                errorCount.inc()
+            }
+        }
+    }
+
+    private fun validEntitySets(entityTypeId: UUID): Set<EntitySet> {
+        return entitySetService.getEntitySetsOfType(entityTypeId)
+                .filter { !it.isLinking && it.flags.contains(EntitySetFlag.TRANSPORTED) }
+                .toSet()
+    }
+
+    @Subscribe
+    fun handleEntityTypeCreated(e: EntityTypeCreatedEvent) {
+        this.syncTable(e.entityType)
+    }
+
+    @Subscribe
+    fun handleAssociationTypeCreated(e: AssociationTypeCreatedEvent) {
+        val entityType = e.associationType.associationEntityType
+        if (entityType != null) {
+            this.syncTable(entityType)
+        }
+    }
+
+    @Subscribe
+    fun handleEntityTypeDeleted(e: EntityTypeDeletedEvent) {
+        this.transporterState.remove(e.entityTypeId)?: return
+        executor.submit {
+            data.datastore().connection.use { conn ->
+                val st = conn.createStatement()
+                st.execute("DROP TABLE ${tableName(e.entityTypeId)}")
+            }
+        }
+    }
+
+//    @Subscribe
```

Comment:
I think we should just delete all this -- `clearAllData` isn't coming back anytime soon

Reply:
👍